A generated Scala project with Gradle doesn't build

**Describe the bug**
When generating a project like so:

```bash
mvn io.quarkus:quarkus-maven-plugin:999-SNAPSHOT:create     \
-DprojectGroupId=org.acme     \
-DprojectArtifactId=gradle-scala-test \     
-DclassName="org.acme.GreetingResource"\    
 -Dpath="/greeting"\
-Dextensions="scala"\ 
-DplatformArtifactId=quarkus-bom \
-DbuildTool=gradle
```

and then trying to build it like so:

```
gradle build -x test
```

I get:

```
> Task :compileScala FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileScala'.
> Cannot infer Scala class path because no Scala library Jar was found. Does root project 'gradle-scala-test' declare dependency to scala-library? Searched classpath: file collection.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 2s
1 actionable task: 1 executed
```

In Windows, a different error is shown in the CI server: https://scans.gradle.com/s/turf2ey7xcl2o/tests/dqwcqqtprpdqc-tvfi6t4p2x5we

**Environment (please complete the following information):**
 - Quarkus version or git rev: latest `master`

--- a/devtools/gradle/src/functionalTest/java/io/quarkus/gradle/QuarkusPluginFunctionalTest.java
+++ b/devtools/gradle/src/functionalTest/java/io/quarkus/gradle/QuarkusPluginFunctionalTest.java
@@ -46,8 +46,7 @@ public class QuarkusPluginFunctionalTest {
     }
 
     @ParameterizedTest(name = "Build {0} project")
-    //TODO: Fix Scala build in Windows
-    @EnumSource(value = SourceType.class, names = {"JAVA","KOTLIN"})
+    @EnumSource(SourceType.class)
     public void canBuild(SourceType sourceType) throws IOException {
         createProject(sourceType);
 

--- a/devtools/gradle/src/functionalTest/java/io/quarkus/gradle/QuarkusPluginFunctionalTest.java
+++ b/devtools/gradle/src/functionalTest/java/io/quarkus/gradle/QuarkusPluginFunctionalTest.java
@@ -5,6 +5,7 @@ import java.io.IOException;
 import java.util.ArrayList;
 import java.util.HashMap;
 import java.util.List;
+import java.util.Map;
 
 import io.quarkus.cli.commands.CreateProject;
 import io.quarkus.cli.commands.writer.FileProjectWriter;
@@ -73,13 +74,16 @@ public class QuarkusPluginFunctionalTest {
     }
 
     private void createProject(SourceType sourceType) throws IOException {
+        Map<String,Object> context = new HashMap<>();
+        context.put("path", "/greeting");
         assertThat(new CreateProject(new FileProjectWriter(projectRoot))
                            .groupId("com.acme.foo")
                            .artifactId("foo")
                            .version("1.0.0-SNAPSHOT")
                            .buildTool(BuildTool.GRADLE)
+                           .className("org.acme.GreetingResource")
                            .sourceType(sourceType)
-                           .doCreateProject(new HashMap<>()))
+                           .doCreateProject(context))
                 .withFailMessage("Project was not created")
                 .isTrue();
     }

--- a/devtools/gradle/src/main/java/io/quarkus/gradle/QuarkusPluginExtension.java
+++ b/devtools/gradle/src/main/java/io/quarkus/gradle/QuarkusPluginExtension.java
@@ -4,6 +4,7 @@ import java.io.File;
 import java.util.Set;
 
 import org.gradle.api.Project;
+import org.gradle.api.file.FileCollection;
 import org.gradle.api.plugins.JavaPluginConvention;
 import org.gradle.api.tasks.SourceSet;
 
@@ -14,66 +15,65 @@ public class QuarkusPluginExtension {
 
     private final Project project;
 
-    private String outputDirectory;
+    private File outputDirectory;
 
     private String finalName;
 
-    private String sourceDir;
+    private File sourceDir;
 
-    private String workingDir;
+    private File workingDir;
 
-    private String outputConfigDirectory;
+    private File outputConfigDirectory;
 
     public QuarkusPluginExtension(Project project) {
         this.project = project;
     }
 
     public File outputDirectory() {
-        if (outputDirectory == null)
-            outputDirectory = project.getConvention().getPlugin(JavaPluginConvention.class)
-                    .getSourceSets().getByName(SourceSet.MAIN_SOURCE_SET_NAME).getOutput().getClassesDirs().getAsPath();
-
-        return new File(outputDirectory);
+        if (outputDirectory == null) {
+            outputDirectory = getLastFile(project.getConvention().getPlugin(JavaPluginConvention.class)
+                    .getSourceSets().getByName(SourceSet.MAIN_SOURCE_SET_NAME).getOutput().getClassesDirs());
+        }
+        return outputDirectory;
     }
 
     public void setOutputDirectory(String outputDirectory) {
-        this.outputDirectory = outputDirectory;
+        this.outputDirectory = new File(outputDirectory);
     }
 
     public File outputConfigDirectory() {
         if (outputConfigDirectory == null) {
             outputConfigDirectory = project.getConvention().getPlugin(JavaPluginConvention.class)
-                    .getSourceSets().getByName(SourceSet.MAIN_SOURCE_SET_NAME).getOutput().getResourcesDir().getAbsolutePath();
+                    .getSourceSets().getByName(SourceSet.MAIN_SOURCE_SET_NAME).getOutput().getResourcesDir();
         }
-        return new File(outputConfigDirectory);
+        return outputConfigDirectory;
     }
 
     public void setOutputConfigDirectory(String outputConfigDirectory) {
-        this.outputConfigDirectory = outputConfigDirectory;
+        this.outputConfigDirectory = new File(outputConfigDirectory);
     }
 
     public File sourceDir() {
         if (sourceDir == null) {
-            sourceDir = project.getConvention().getPlugin(JavaPluginConvention.class)
-                    .getSourceSets().getByName(SourceSet.MAIN_SOURCE_SET_NAME).getAllJava().getSourceDirectories().getAsPath();
+            sourceDir = getLastFile(project.getConvention().getPlugin(JavaPluginConvention.class)
+                    .getSourceSets().getByName(SourceSet.MAIN_SOURCE_SET_NAME).getAllJava().getSourceDirectories());
         }
-        return new File(sourceDir);
+        return sourceDir;
     }
 
     public void setSourceDir(String sourceDir) {
-        this.sourceDir = sourceDir;
+        this.sourceDir = new File(sourceDir);
     }
 
     public File workingDir() {
         if (workingDir == null) {
-            workingDir = outputDirectory().getPath();
+            workingDir = outputDirectory();
         }
-
-        return new File(workingDir);
+        return workingDir;
     }
 
     public void setWorkingDir(String workingDir) {
-        this.workingDir = workingDir;
+        this.workingDir = new File(workingDir);
     }
 
     public String finalName() {
@@ -99,5 +99,17 @@ public class QuarkusPluginExtension {
 
     public AppModelResolver resolveAppModel() {
         return new AppModelGradleResolver(project);
+    }
+
+    /**
+     * Returns the last file from the specified {@link FileCollection}.
+     * Needed for the Scala plugin.
+     */
+    private File getLastFile(FileCollection fileCollection) {
+        File result = null;
+        for (File f : fileCollection) {
+            result = f;
+        }
+        return result;
     }
 }

--- a/devtools/platform-descriptor-json/src/main/resources/templates/basic-rest/scala/build.gradle-template.ftl
+++ b/devtools/platform-descriptor-json/src/main/resources/templates/basic-rest/scala/build.gradle-template.ftl
@@ -10,6 +10,7 @@ repositories {
 
 dependencies {
     implementation enforcedPlatform("${quarkusPlatformGroupId}:${quarkusPlatformArtifactId}:${quarkusPlatformVersion}")
+    implementation 'org.scala-lang:scala-library:${scala_version}'
     implementation 'io.quarkus:quarkus-resteasy'
 
     testImplementation 'io.quarkus:quarkus-junit5'

